src/store/student-store.js

```diff
@@ -18,13 +18,8 @@ class StudentStore extends Store{
 const store = new StudentStore();
 
 store.registerHandler('STUDENT_FOUND', data => {
-    student = {
-        //NOTE: this data is tentative
-        id : data.id,
-        name: data.name,
-        items: data.items,
-        hasOverdueItem: store.hasOverdueItems(data.items)
-    };
+    student = data;
+    student.hasOverdueItem = store.hasOverdueItems(data.items);
```

Comment:
What? Why was this not brought up in the code review that this was implemented for? You should talk to @Cafferoq for that as I think he implemented this not I.